Exporter/src/main/java/uk/ac/ebi/biostd/exporter/jobs/pmc/importer/process/SubmissionsParser.java

```diff
@@ -0,0 +1,94 @@
+package uk.ac.ebi.biostd.exporter.jobs.pmc.importer.process;
+
+import static java.lang.String.format;
+import static uk.ac.ebi.biostd.model.Submission.canonicReleaseDateAttribute;
+
+import com.google.common.collect.Lists;
+import java.io.File;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.biostd.in.pagetab.SubmissionInfo;
+import uk.ac.ebi.biostd.model.Section;
+import uk.ac.ebi.biostd.model.SectionAttribute;
+import uk.ac.ebi.biostd.model.Submission;
+import uk.ac.ebi.biostd.treelog.ErrorCounterImpl;
+import uk.ac.ebi.biostd.treelog.LogNode.Level;
+import uk.ac.ebi.biostd.treelog.SimpleLogNode;
+
+@Slf4j
+@AllArgsConstructor
+@Component
+public class SubmissionsParser {
+
+    public static final String PUB_DATE_PARSING_ERROR = "Did not find year for submission '%s' in publication date "
+            + "'%s'";
+
+    private final CvsTvsParser cvsTvsParser;
+
+    public List<SubmissionInfo> parseSubmissionFile(File file) {
+        SimpleLogNode topLn = new SimpleLogNode(Level.SUCCESS, "Parsing file: ", new ErrorCounterImpl());
+        List<SubmissionInfo> submissions = cvsTvsParser.parseZipFile(file, '\t', topLn).getSubmissions();
+        return removeDuplicated(submissions);
+    }
+
+    private List<SubmissionInfo> removeDuplicated(List<SubmissionInfo> submissions) {
+        Set<String> submissionSet = new HashSet<>();
+        List<SubmissionInfo> uniqueSubmissions = new ArrayList<>();
+
+        for (SubmissionInfo info : Lists.reverse(submissions)) {
+            if (isNew(submissionSet, info.getAccNoOriginal())) {
+                uniqueSubmissions.add(info);
+                submissionSet.add(info.getAccNoOriginal());
+            }
+        }
+
+        return setReleaseDate(uniqueSubmissions);
+    }
+
+    @SneakyThrows
+    private List<SubmissionInfo> setReleaseDate(List<SubmissionInfo> submissions) {
+        for (SubmissionInfo submissionInfo : submissions) {
+            Submission sub = submissionInfo.getSubmission();
+
+            for (Section section : sub.getRootSection().getSections()) {
+                if (!section.getType().equalsIgnoreCase("Publication")) {
+                    continue;
+                }
+
+                for (SectionAttribute attr : section.getAttributes()) {
+                    if (attr.getName().equalsIgnoreCase("Publication date")) {
+                        sub.addAttribute(canonicReleaseDateAttribute, tryToParse(attr.getValue(), sub.getAccNo()));
+                    }
+                }
+            }
+        }
+
+        return submissions;
```

Comment:
At first glance this comes to mind:
```java
submissions.forEach(submissionInfo -> {
    submissionInfo
        .getSubmission().getRootSection().getSections()
        .filter(section -> section.getType().equalsIgnoreCase("Publication"))
        .forEach(section -> {
            section
                .getAttributes()
                .filter(attr -> attr.getName().equalsIgnoreCase("Publication date"))
                .forEach(attr -> sub.addAttribute(canonicReleaseDateAttribute, tryToParse(attr.getValue(), sub.getAccNo()));
        });
});
```

WDYT?

Reply:
even I do like a lot lamdas and prefer to regular bucles , this  one looks quite complex and IMHO do not make the code cleaner.

Reply:
Ok